src/components-v2/vault-detail/VaultDetail.tsx

```diff
@@ -27,26 +26,20 @@ const useStyles = makeStyles((theme) => ({
 		textAlign: 'center',
 		marginTop: theme.spacing(10),
 	},
+	notFoundImage: {
+		marginTop: theme.spacing(10),
+	},
 }));
 
 export const VaultDetail = observer((): JSX.Element => {
-	const {
-		vaultDetail,
-		network: { network },
-		router,
-		vaults,
-	} = useContext(StoreContext);
-
-	const initialNetwork = useRef(network);
+	const { vaultDetail, vaults, router } = useContext(StoreContext);
 	const classes = useStyles();
 	const { vault, isLoading, isNotFound, isDepositDialogDisplayed, isWithdrawDialogDisplayed } = vaultDetail;
 	const badgerVault = vault ? vaults.getVaultDefinition(vault) : undefined;
 
-	useEffect(() => {
-		if (network.symbol !== initialNetwork.current.symbol) {
-			router.goTo(routes.home);
-		}
-	}, [network, router]);
```

Comment:
it turns out that this **explicit** redirect was the root cause of the redirects 😃